desktop_version/src/Music.cpp

```diff
@@ -92,81 +131,263 @@ class SoundTrack
+        if (!IsHalted()) FAudioVoice_SetVolume(musicVoice, adj_vol, FAUDIO_COMMIT_NOW);
     }
 
 private:
-    Mix_Music *m_music;
-};
+    stb_vorbis* vorbis = NULL;
+    int channels;
+    Uint32 size;
+    int loopbegin;
+    int looplength;
+    int sample_pos; //stb_vorbis offset not yet functional on pulldata API. TODO Replace when fixed
+
+    FAudioVoiceCallback callbacks;
+    FAudioWaveFormatEx format;
+
+    Uint8* decoded_buf_playing = NULL;
+    Uint8* decoded_buf_reserve = NULL;
+    Uint8* read_buf = NULL;
+    bool shouldloop;
+
+    static FAudioSourceVoice* musicVoice;
+
+    static void refillReserve(FAudioVoiceCallback* callback, void* ctx) {
+        MusicTrack* t = (MusicTrack*)ctx;
+        FAudioBuffer faudio_buffer;
+        SDL_memset(&faudio_buffer, 0, sizeof(FAudioBuffer));
+        if (t->looplength == 0) {
+            faudio_buffer.PlayLength = stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float));
+        } else {
+            faudio_buffer.PlayLength = std::min(stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float)), (t->loopbegin+t->looplength)-t->sample_pos);
+        }
+        faudio_buffer.AudioBytes = t->size;
+        faudio_buffer.pAudioData = t->decoded_buf_reserve;
+        faudio_buffer.pContext = t;
+        if (faudio_buffer.PlayLength == 0) {
+            if (t->shouldloop) {
+                stb_vorbis_seek(t->vorbis, t->loopbegin);
+                t->sample_pos = t->loopbegin;
+                if (t->looplength != 0) {
+                    faudio_buffer.PlayLength = std::min(stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float)), t->looplength);
+                } else {
+                    faudio_buffer.PlayLength = stb_vorbis_get_samples_float_interleaved(t->vorbis, t->channels, (float*)t->decoded_buf_reserve, t->size/sizeof(float));
+                }
+                if (faudio_buffer.PlayLength == 0) return;
+            } else {
+                return;
+            }
+        }
+        t->sample_pos += faudio_buffer.PlayLength;
+        FAudioSourceVoice_SubmitSourceBuffer(musicVoice, &faudio_buffer, NULL);
+    }
+    static void swapBuffers(FAudioVoiceCallback* callback, void* ctx) {
+        MusicTrack* t = (MusicTrack*)ctx;
+        std::swap(t->decoded_buf_playing, t->decoded_buf_reserve);
+    }
+
+    //Lifted from sdl_mixer
+    static void parseComments(MusicTrack* t, char** comments, int comment_list_length) {
+        int loopend = 0;
+        for (int i = 0; i < comment_list_length; i++) {
+            char *param = SDL_strdup(comments[i]);
+            char *argument = param;
+            char *value = SDL_strchr(param, '=');
+            if (value == NULL) {
+                value = param + SDL_strlen(param);
+            } else {
```

Comment:
Closing brace needs to be on a separate line. Opening brace needs to be on the next line.